.github/workflows/release.yaml

```diff
@@ -0,0 +1,121 @@
+      - name: Validate input tag
+        run: ./semver validate $SEMVER_TAG
+      - name: Prevent major/minor version change
+        run: |
+          source VERSION
+          diff=$(./semver diff "$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH$VERSION_SUFFIX" $SEMVER_TAG)
+          if [[ "$diff" == "prerelease" ]]
+          then
+            echo "version match"
+            exit 0
+          fi
+
+          # prevent changes on major/minor/patch version
+          # major/minor these are done manually by creating a new dev/* branch
+          # patch is done at the end of this workflow
+          echo "version mismatch, can only upgrade patch version"
+          exit 1
+      - name: Check version is on the right branch
+        run: |
+          source VERSION
+
+          # Need to unshallow and fetch dev branches first
+          git fetch --unshallow --no-tags origin HEAD '+refs/heads/development/*:refs/remotes/origin/development/*'
+
+          GIT_REMOTE=$(git remote)
+          BASE_BRANCH=$(git branch -a --list ${GIT_REMOTE}/development/${VERSION_MAJOR}.${VERSION_MINOR})
+          if ! git merge-base --is-ancestor HEAD ${BASE_BRANCH}; then
+            echo "Version commit is not included in base branch ${BASE_BRANCH}"
+            exit 1
+          fi
+
+          NEXT_MAJOR=$((VERSION_MAJOR+1)).0
+          NEXT_MINOR=${VERSION_MAJOR}.$((VERSION_MINOR+1))
+          NEXT_PATCH=${VERSION_MAJOR}.${VERSION_MINOR}.$((VERSION_PATCH+1))
+          for BRANCH in $(git branch -a --list ${GIT_REMOTE}/development/{$NEXT_MAJOR,$NEXT_MINOR,$NEXT_PATCH}); do
+            if git merge-base --is-ancestor $BRANCH HEAD ; then
+              echo "Version commit includes next target branch ${BRANCH}"
+              exit 1
+            fi
+          done
+      - name: Prepare branch
+        run: git checkout -b ${{env.RELEASE_BRANCH}}
+      - name: Set the new version
+        run: |
+          if [ -z "${NEW_VERSION_SUFFIX}" ]; then
+            NEW_VERSION_SUFFIX=$(./semver get prerel ${{inputs.tag}})
+            [ -n "$NEW_VERSION_SUFFIX" ] && NEW_VERSION_SUFFIX="-$NEW_VERSION_SUFFIX"
+          fi
+          sed -i "s/VERSION_SUFFIX=.*/VERSION_SUFFIX=$NEW_VERSION_SUFFIX/" VERSION
+      - name: Commit with requested version and push
+        run: |
+          git fsck
+          git gc
+          git add VERSION
+          git config --global user.email ${{github.actor}}@scality.com
+          git config --global user.name ${{github.actor}}
+          git commit -m "Release version ${{inputs.tag}}"
+          git push --set-upstream origin ${{env.RELEASE_BRANCH}}
+      - name: Create Release
+        uses: softprops/action-gh-release@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GIT_ACCESS_TOKEN }}
+        with:
+          name: Release ${{ inputs.tag }}
+          tag_name: ${{ inputs.tag }}
+          generate_release_notes: true
+          target_commitish: ${{ env.RELEASE_BRANCH }}
```

Comment:
No, here we only want to tag not create the release on github, the release creation is done by the promote step

Reply:
Removed!